server/server/server.go

```diff
@@ -1,38 +1,98 @@
 package server
 
 import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
 	"github.com/gin-gonic/gin"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/newrelic/go-agent/v3/integrations/nrgrpc"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_zap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
 	"github.com/odpf/stencil/server/api"
+	"github.com/odpf/stencil/server/api/v1/pb"
 	"github.com/odpf/stencil/server/config"
+	"github.com/odpf/stencil/server/logger"
 	"github.com/odpf/stencil/server/proto"
 	"github.com/odpf/stencil/server/snapshot"
 	"github.com/odpf/stencil/server/store"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	"google.golang.org/grpc"
 )
 
 // Router returns server router
-func Router(api *api.API, config *config.Config) *gin.Engine {
+func Router(api *api.API, config *config.Config) *runtime.ServeMux {
+	gwmux := runtime.NewServeMux()
 	router := gin.New()
 	addMiddleware(router, config)
 	registerCustomValidations(router)
-	registerRoutes(router, api)
-	return router
+	registerRoutes(router, gwmux, api)
+	return gwmux
 }
 
 // Start Entry point to start the server
 func Start() {
+	ctx := context.Background()
 	config := config.LoadConfig()
 	db := store.NewDBStore(config)
 	stRepo := snapshot.NewSnapshotRepository(db)
-	stSvc := snapshot.NewSnapshotService(stRepo)
 	protoRepo := proto.NewProtoRepository(db)
 	protoService := proto.NewService(protoRepo, stRepo)
 	api := &api.API{
 		Store:    protoService,
-		Metadata: stSvc,
+		Metadata: stRepo,
+	}
+	port := fmt.Sprintf(":%s", config.Port)
+	nr := getNewRelic(config)
+	mux := Router(api, config)
+
+	// init grpc server
+	opts := []grpc.ServerOption{
+		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
+			grpc_recovery.UnaryServerInterceptor(),
+			grpc_ctxtags.UnaryServerInterceptor(),
+			nrgrpc.UnaryServerInterceptor(nr),
+			grpc_zap.UnaryServerInterceptor(logger.Logger))),
+		grpc.MaxRecvMsgSize(config.GRPCMaxRecvMsgSizeInMB << 20),
+		grpc.MaxSendMsgSize(config.GRPCMaxSendMsgSizeInMB << 20),
 	}
-	router := Router(api, config)
+	// Create a gRPC server object
+	s := grpc.NewServer(opts...)
+	pb.RegisterStencilServiceServer(s, api)
+	conn, err := grpc.DialContext(
+		context.Background(),
+		fmt.Sprintf("0.0.0.0%s", port),
```

Comment:
I think if we just gonna run on `0.0.0.0`, we can simply pass `port` variable as well, that is, `:8080`.

Reply:
I didn't get this. What do you mean?

Reply:
```go
conn, err := grpc.DialContext(
    context.Background(),
    port,
    grpc.WithInsecure(),
)
```
I think this is equivalent to what you wrote.